ola/ClientTypes.cpp

```diff
@@ -0,0 +1,125 @@
+ * ClientTypes.cpp
+ * Copyright (C) 2015 Simon Marchi
+ */
+
+#include <vector>
+
+#include "ola/client/ClientTypes.h"
+
+namespace ola {
+namespace client {
+
+using std::vector;
+
+/*
+ * Create an OlaPlugin object from a protobuf.
+ */
+OlaPlugin OlaPlugin::FromProtobuf(ola::proto::PluginInfo plugin_info) {
+  return OlaPlugin(plugin_info.plugin_id(),
+                   plugin_info.name(),
+                   plugin_info.active(),
+                   plugin_info.enabled());
+}
+
+/*
+ * Create an OlaInputPort object from a protobuf.
+ */
+OlaInputPort OlaInputPort::FromProtobuf(ola::proto::PortInfo port_info) {
+  return OlaInputPort(port_info.port_id(),
+                      port_info.universe(),
+                      port_info.active(),
+                      port_info.description(),
+                      static_cast<port_priority_capability>(
+                          port_info.priority_capability()),
+                      static_cast<port_priority_mode>(
+                          port_info.priority_mode()),
+                      port_info.priority(),
+                      port_info.supports_rdm());
+}
+
+/*
+ * Create an OlaOutputPort object from a protobuf.
+ */
+OlaOutputPort OlaOutputPort::FromProtobuf(ola::proto::PortInfo port_info) {
+  return OlaOutputPort(port_info.port_id(),
+                       port_info.universe(),
+                       port_info.active(),
+                       port_info.description(),
+                       static_cast<port_priority_capability>(
+                           port_info.priority_capability()),
+                       static_cast<port_priority_mode>(
+                           port_info.priority_mode()),
+                       port_info.priority(),
+                       port_info.supports_rdm());
+}
+
+/*
+ * Create an OlaDevice object from a protobuf.
+ */
+OlaDevice OlaDevice::FromProtobuf(ola::proto::DeviceInfo device_info) {
+  vector<OlaInputPort> input_ports;
+  for (int i = 0; i < device_info.input_port_size(); ++i) {
+    ola::proto::PortInfo port_info = device_info.input_port(i);
+    input_ports.push_back(OlaInputPort::FromProtobuf(port_info));
+  }
+
+  vector<OlaOutputPort> output_ports;
+  for (int i = 0; i < device_info.output_port_size(); ++i) {
+    ola::proto::PortInfo port_info = device_info.output_port(i);
+    output_ports.push_back(OlaOutputPort::FromProtobuf(port_info));
+  }
+
+  return OlaDevice(device_info.device_id(),
+                   device_info.device_alias(),
+                   device_info.device_name(),
+                   device_info.plugin_id(),
+                   input_ports,
+                   output_ports);
+}
+
+/*
+ * Create an OlaUniverse object from a protobuf.
+ */
+OlaUniverse OlaUniverse::FromProtobuf(
+    ola::proto::UniverseInfo universe_info) {
+  OlaUniverse::merge_mode merge_mode =
+    universe_info.merge_mode() == ola::proto::HTP ?
+    OlaUniverse::MERGE_HTP: OlaUniverse::MERGE_LTP;
+
+  vector<OlaInputPort> input_ports;
```

Comment:
If you transfer ownership ,  you can make these pointers:

vector<OlaInputPort> *input_ports = new vector<OlaInputPort>();